client.js

```diff
@@ -117,15 +117,19 @@ function EventSourceWrapper() {
 }
 
 function getEventSourceWrapper() {
-  if (!window.__whmEventSourceWrapper) {
-    window.__whmEventSourceWrapper = {};
+  var top = window;
+  if (window.self !== window.top) {
+    top = window.top;
   }
-  if (!window.__whmEventSourceWrapper[options.path]) {
+  if (!top.__whmEventSourceWrapper) {
+    top.__whmEventSourceWrapper = {};
+  }
+  if (!top.__whmEventSourceWrapper[options.path]) {
     // cache the wrapper for other entries loaded on
     // the same page with the same options.path
-    window.__whmEventSourceWrapper[options.path] = EventSourceWrapper();
+    top.__whmEventSourceWrapper[options.path] = EventSourceWrapper();
```

Comment:
I think we should not keep it cached now. e.g. On this reload server is closing web socket connection 
`module.hot.accept() => {
            window.location.reload();
 });`
while when creating new EventSourceWrapper(); it work properly.